plugins/modules/redhat_subscription.py

```diff
@@ -430,6 +432,24 @@ def register(self, was_registered, username, password, token, auto_attach, activ
             Raises:
               * Exception - if any error occurs during the registration
         '''
+
+        def str2int(s, default=0):
+            try:
+                return int(s)
+            except ValueError:
+                return default
+
+        distro_id = distro.id()
+        distro_version_parts = distro.version_parts()
+        distro_version = tuple(str2int(p) for p in distro_version_parts)
+
+        # subscription-manager attach command was removed in Fedora 41 and RHEL 10
+        if (
+            (distro_id == 'rhel' and distro_version[0] >= 10)
+            or (distro_id == 'fedora' and distro_version[0] >= 41)
+        ):
+            auto_attach = False
```

Comment:
Assuming that this module will only work for distros with a numeric version number (dotted or not), this could be made simpler:
```suggestion
        distro_version = int(distro.version_parts().split('.')[0])

        # subscription-manager attach command was removed in Fedora 41 and RHEL 10
        if (distro_id == 'rhel' and distro_version >= 10)
            or (distro_id == 'fedora' and distro_version >= 41):
            auto_attach = False
```

Reply:
I'll be honest, I copy-pasted much of the version number handling from code added in https://github.com/ansible-collections/community.general/pull/7624. I agree it's overkill though, since the change I'm adding here doesn't need to inspect minor version number at all. While poking around in the `distro` module utility I noticed that there's a `major_version` method, so I've updated the PR to fetch the major version only.

I did opt to keep the exception handling for `ValueError` though, to keep the module from failing badly if it's ever run on a system with a letter in the major version. Probably only likely to happen if someone uses the module with a completely inappropriate OS, but I think it's better in that case for the module to fail with an error like 'subscription-manager: command not found` instead of `ValueError`.